File: backend/src/python/Dockerfile.py
# Use an official Python runtime as a base image
FROM python:3.11

# Set the working directory inside the container
WORKDIR /backend/src/python

# Copy the Python client script into the container
COPY backend/src/python/tcp_c.py .

# Expose the TCP client port (not always required, but useful for debugging)
EXPOSE 12345

# Default command to run the Python client
CMD ["python3", "tcp_c.py", "tcp-server", "12345"]

File: backend/Dockerfile.js
# Use the official Node.js runtime as the base image
FROM node:16

# Set the correct working directory
WORKDIR /NETFLIX-project/backend

# Copy package.json and package-lock.json first (for caching)
COPY backend/package*.json ./

# Run npm install before copying the rest of the app
RUN npm install

# Copy the rest of the backend source code
COPY backend/. .

# Expose the port your app uses (3000 in this case)
EXPOSE 3000

# Run the Node application (assuming main file is `app.js`)
CMD ["node", "src/app.js"]